performance_metrics/tutorial.md

```diff
@@ -0,0 +1,142 @@
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor"
+  real_sensor_update_rate: 235.6538640635398
+  sim_sensor_update_rate: 250
+  fps: 241.61639404296875
+}
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor_regular"
+  real_sensor_update_rate: 31.1463905584825
+  sim_sensor_update_rate: 30.3030303030303
+  fps: 29.975315093994141
+}
+```
+
+### Run the world with lockstep
+
+With the `lockstep` flag activated the `sim_sensor_update_rate` must correspond with the real value
+defined in the world. In this case we expect a `sim_sensor_update_rate` equals to 500 in the high resolution camera
+and 30 in the low resolution. Then the `fps` value means that Gazebo is able to generate 337 frames in a *real second*.
+
+The real time factor is likely less than 1.0. The exact number depends on your computing power. This shows that
+the sensor's update rate is strictly followed, and physics has slowed down in order to accommodate for the high update rate.
+
+```bash
+real_time_factor: 0.67489652
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor"
+  real_sensor_update_rate: 348.7102949740385
+  sim_sensor_update_rate: 500
+  fps: 337.46530151367188
+}
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor_regular"
+  real_sensor_update_rate: 21.522750989820555
+  sim_sensor_update_rate: 30.3030303030303
+  fps: 20.173301696777344
+}
+```
+
+### Real time update rate
+
+We can modify the Real time update rate in the Physics Engine. As you may see the `sim_sensor_update_rate` in both cameras
+are the expected value 500 and 30. In this case the Real time update rate is equal to **10** which make the simulation slower. The fps
+are lower, it's able to generate almost 6 frames for the high speed camera and 2 for the regular camera each *real second*.
+
+#### Run the world with lockstep
+
+```bash
+real_time_factor: 0.67666977300000009
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor"
+  real_sensor_update_rate: 5.0383346478096511
+  sim_sensor_update_rate: 500
+  fps: 5.61544132232666
+}
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor_regular"
+  real_sensor_update_rate: 0.29411702534733497
+  sim_sensor_update_rate: 29.411764705882351
+  fps: 2.0637693405151367
+}
+```
+
+#### Run the world without lockstep
+
+Again in this case, where we run the world without `lockstep`, there is no guarantee when the callback is called,
+and the `sim_sensor_update_rate` value is highly impacted, it should be close to 500 but in this case it's close to 0.
+
+```bash
+real_time_factor: 0.425736361
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor"
+  real_sensor_update_rate: 96.478915160204693
+  sim_sensor_update_rate: 0.14425851125216388
+  fps: 80.7220458984375
+}
+sensor {
+  sensor_name: "default::camera_model::link::camera_sensor_regular"
+  real_sensor_update_rate: 169.40730989154036
+  sim_sensor_update_rate: 0.14452955629426217
```

Comment:
interesting. If we slow down physics by changing the `real_time_update_rate` to `10`, the camera sensor should have more time to catch up and hit it's target rate

I'm seeing quite different values from the ones here using the latest changes that compute metrics inside SensorManager.

The `real_sensor_update_rate` for me is close to ~0.3, while `sim_sensor_update_rate` is ~30. Same for the high speed camera above, I'm getting close to ~5 `real_sensor_update_rate` and ~500 `sim_sensor_update_rate`

Reply:
Added instantaneous RTF in the PR : https://github.com/osrf/gazebo/pull/2819#discussion_r481971068